apps/obsidian/src/components/RelationshipSettings.tsx

```diff
@@ -0,0 +1,296 @@
+import { useState } from "react";
+import {
+  DiscourseRelation,
+  DiscourseNode,
+  DiscourseRelationType,
+} from "../types";
+import { Notice } from "obsidian";
+import { usePlugin } from "./PluginContext";
+
+const RelationshipSettings = () => {
+  const plugin = usePlugin();
+  const [discourseRelations, setDiscourseRelations] = useState<
+    DiscourseRelation[]
+  >(() => plugin.settings.discourseRelations ?? []);
+  const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
+
+  const findNodeById = (id: string): DiscourseNode | undefined => {
+    return plugin.settings.nodeTypes.find((node) => node.id === id);
+  };
+
+  const findRelationTypeById = (
+    id: string,
+  ): DiscourseRelationType | undefined => {
+    return plugin.settings.relationTypes.find((relType) => relType.id === id);
+  };
+
+  const handleRelationChange = async (
+    index: number,
+    field: keyof DiscourseRelation,
+    value: string,
+  ): Promise<void> => {
+    const updatedRelations = [...discourseRelations];
+
+    if (!updatedRelations[index]) {
+      updatedRelations[index] = {
+        sourceId: "",
+        destinationId: "",
+        relationshipTypeId: "",
+      };
+    }
+
+    updatedRelations[index][field] = value;
+    setDiscourseRelations(updatedRelations);
+    setHasUnsavedChanges(true);
+  };
+
+  const handleAddRelation = (): void => {
+    const updatedRelations = [
+      ...discourseRelations,
+      {
+        sourceId: "",
+        destinationId: "",
+        relationshipTypeId: "",
+      },
+    ];
+    setDiscourseRelations(updatedRelations);
+    setHasUnsavedChanges(true);
+  };
+
+  const handleDeleteRelation = async (index: number): Promise<void> => {
```

Comment:
nit - we can address this later
since we are using the unsaved changes pattern, we should probably also get confirmation before deleting